### PR TITLE
use extendEnv instead of append to container.Env

### PR DESCRIPTION
**Pull request by nicmorales9**
# Description

this helps avoid collisions in the case that a variable is declared in the cluster spec already (collisions will prevent Patch()s and prevent Operator from functioning normally)

## Type of change

- minor improvement

## Discussion

*Are there any design details that you would like to discuss further?*

## Testing

*Please describe the tests that you ran to verify your changes. Unit tests?
Manual testing?*

*Do we need to perform additional testing once this is merged, or perform in a larger testing environment?*

## Documentation

*Did you update relevant documentation within this repository?*

*If this change is adding new functionality, do we need to describe it in our user manual?*

*If this change is adding or removing subreconcilers, have we updated the core technical design doc to reflect that?*

*If this change is adding new safety checks or new potential failure modes, have we documented and how to debug potential issues?*

## Follow-up

*Are there any follow-up issues that we should pursue in the future?*

*Does this introduce new defaults that we should re-evaluate in the future?*

### CODE REVIEW

File: internal/pod_models_test.go [hunk 1]
```diff
@@ -512,7 +512,7 @@ var _ = Describe("pod_models", func() {
 						"--log-path", "/var/log/fdb-trace-logs/monitor.log",
 					}))
 
-					Expect(mainContainer.Env).To(Equal([]corev1.EnvVar{
+					Expect(mainContainer.Env).To(ContainElements([]corev1.EnvVar{
```
Comment (johscheuer) [inline internal/pod_models_test.go:515]: What was the reasoning behind this change? I believe if we want to change that it should be `ConsistOf` to make sure that exactly those env variables are present (without caring about the order).

File: internal/pod_models.go [hunk 2]
```diff
@@ -199,19 +200,22 @@ func configureContainersForUnifiedImages(cluster *fdbv1beta2.FoundationDBCluster
...
+		}},
+		corev1.EnvVar{Name: "FDB_NETWORK_OPTION_TRACE_LOG_GROUP", Value: cluster.GetLogGroup()},
+		corev1.EnvVar{Name: "FDB_NETWORK_OPTION_TRACE_ENABLE", Value: "/var/log/fdb-trace-logs"},
+	)
+	mainContainerEnv = append(mainContainerEnv, getEnvForMonitorConfigSubstitution(cluster, processGroup.ProcessGroupID)...)
 	if cluster.DefineDNSLocalityFields() {
-		mainContainer.Env = append(mainContainer.Env, corev1.EnvVar{Name: "FDB_DNS_NAME", Value: GetPodDNSName(cluster, processGroup.GetPodName(cluster))})
+		mainContainerEnv = append(mainContainerEnv, corev1.EnvVar{Name: "FDB_DNS_NAME", Value: GetPodDNSName(cluster, processGroup.GetPodName(cluster))})
 	}
+	extendEnv(mainContainer, mainContainerEnv...)
```
Comment (johscheuer) [inline internal/pod_models.go:218]: I noticed that the `extendEnv ` has a bug where a user could add the same env multiple times in the same `extendEnv` call. The fix is pretty simple, we have to add here: https://github.com/FoundationDB/fdb-kubernetes-operator/blob/main/internal/pod_models.go#L810 (in the if statement) `existingVars[envVar.Name] = true`.

File: internal/pod_models_test.go (hunk 1 above)
Comment (nicmorales9) [reply internal/pod_models_test.go:515]: the new method will change the cap and cause `Equal()` checks to fail, I'll swap to `ConsistOf` no problem

File: internal/pod_models.go (hunk 2 above)
Comment (nicmorales9) [reply internal/pod_models.go:218]: ooh nice catch

**Review by johscheuer — APPROVED** [review]
LGTM 👍

**Review by johscheuer — APPROVED** [review]
LGTM 👍

**Comment by johscheuer** [comment]
> ### Result of fdb-kubernetes-operator-pr on Linux CentOS 7
> * Commit ID: [2a03693](https://github.com/FoundationDB/fdb-kubernetes-operator/commit/2a03693fc0c2d3ed254c68cad884072916989a1b)
> * Duration 2:38:48
> * Result: ❌ **FAILED**
> * Error: `Error while executing command: if $fail_test; then exit 1; fi. Reason: exit status 1`
> * [Build Log](https://d1e0l78xbkh3xa.cloudfront.net/fdb-kubernetes-operator/fdb-kubernetes-operator-pr/6a52ca2e-69cb-4e44-b5fa-919269ba67cf.gz) terminal output (available for 30 days)
> * [Build Workspace](https://d1e0l78xbkh3xa.cloudfront.net/fdb-kubernetes-operator-pr/6a52ca2e-69cb-4e44-b5fa-919269ba67cf/workspace.zip) zip file of the working directory (available for 30 days)

Failure is unrelated.

**Pull request merged**